src/components/Tabs.js

```diff
@@ -0,0 +1,203 @@
+const React = require('react');
+const Radium = require('radium');
+
+const Icon = require('./Icon');
+const SimpleSelect = require('./SimpleSelect');
+const StyleConstants = require('../constants/Style');
+
+const Tabs = React.createClass({
+  propTypes: {
+    brandColor: React.PropTypes.string,
+    onTabSelect: React.PropTypes.func.isRequired,
+    selectedBottomBorderPadding: React.PropTypes.number,
+    selectedTab: React.PropTypes.number,
+    showBottomBorder: React.PropTypes.bool,
+    tabs: React.PropTypes.array.isRequired
+  },
+
+  getDefaultProps () {
+    return {
+      brandColor: StyleConstants.Colors.PRIMARY,
+      showBottomBorder: true,
+      selectedBottomBorderPadding: 5
+    };
+  },
+
+  getInitialState () {
+    return {
+      selectedTab: this.props.selectedTab || 0,
+      showMenu: false
+    };
+  },
+
+  componentWillReceiveProps (nextProps) {
+    if (nextProps.selectedTab !== this.state.selectedTab) {
+      this.setState({
+        selectedTab: nextProps.selectedTab
+      });
+    }
+  },
+
+  _handleMenuClick () {
+    this.setState({
+      showMenu: !this.state.showMenu
+    });
+  },
+
+  _handleTabClick (selectedTab) {
+    this.props.onTabSelect(selectedTab);
+    this._handleMenuClick();
+
+    this.setState({
+      selectedTab
+    });
+  },
+
+  _isLargeOrMediumWindowSize () {
+    const windowSize = StyleConstants.getWindowSize();
+
+    return windowSize === 'medium' || windowSize === 'large';
+  },
+
+  _renderTabs () {
+    const selectedTabStyle = {
+      borderBottom: '2px solid ' + this.props.brandColor,
+      paddingBottom: this.props.selectedBottomBorderPadding
+    };
+    const styles = this.styles();
+
+    return this.props.tabs.map((tab, index) => {
+      const _index = index;
+
+      return (
+        <span
+          key={_index}
+          onClick={this._handleTabClick.bind(null, _index)}
+          style={[styles.tab, this.state.selectedTab === _index && selectedTabStyle]}
+        >
+          {tab}
+        </span>
+      );
+    });
+  },
+
+  _renderTabMenu () {
+    const selectedTab = this.props.tabs[this.state.selectedTab];
+    const styles = this.styles();
+    const tabItems = this._tabItems();
+
+    return (
+      <div onClick={this._handleMenuClick} style={styles.menuWrapper} >
+        {selectedTab}
+        <Icon
+          size={20}
+          style={{ color: this.state.brandColor }}
+          type={!this.state.showMenu ? 'caret-down' : 'caret-up' }
+        />
+        {this.state.showMenu ? (
+          <SimpleSelect
+            items={tabItems}
+            menuStyles={styles.menu}
+            onScrimClick={this._handleMenuClick}
+            showItems={this.state.showMenu}
+          />
+        ) : null}
+      </div>
+    );
+  },
+
+  _tabItems () {
```

Comment:
`renderTabMenu` should be in charge of rendering the items, but not building the `_tabItems`. That's why I split them out. Splitting these out also makes it more discoverable for anyone jumping in.